docs/installation.rst

```diff
@@ -1,25 +1,18 @@
 Installation
 ============
 
-The Neovim Python client supports Python 2.7, and 3.4 or later.
+The Neovim Python client supports Python 3.6 or later.
```

Comment:
3.6 is end-of-life now, too (since Dec 23, 2021), so might as well bump to 3.7 (which still has till June 27, 2023).

Reply:
Bumped. Note that I left the `python_requires=">=3.6"` in the setup.py because of CI issues. The latest pypy3 in Ubuntu repos is only compatible with Python 3.6, so if we want to bump the `python_requires` we have to either stop testing on pypy3 (probably undesired) or wrangle the CI environment to install a more recent pypy3. I think that's outside of the scope of this PR.

We probably want to move to github workflows anyway, since travis-ci got rid of their easy-to-use free tier for open source projects.

Reply:
Yeah, looks like travis is already broken because it depends on https://github.com/neovim/bot-ci, which was archived.